pkg/console/controllers/clidownloads/controller.go

```diff
@@ -114,8 +121,27 @@ func (c *CLIDownloadsSyncController) Sync(ctx context.Context, controllerContext
 	}
 
 	statusHandler := status.NewStatusHandler(c.operatorClient)
+
+	infrastructureConfig, err := c.infrastructureClient.Get(ctx, api.ConfigResourceName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("infrastructure config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	clusterVersionConfig, err := c.clusterVersionLister.Get("version")
+	if err != nil {
+		klog.Errorf("cluster version config error: %v", err)
+		return statusHandler.FlushAndReturn(err)
+	}
+
+	// Disable the client download check for external control plane topology (hypershift) if the ingress capability is disabled.
+	if controllersutil.IsExternalControlPlaneWithIngressDisabled(infrastructureConfig, clusterVersionConfig) {
```

Comment:
by doing this you are basically disabling the operator from doing its work by syncing the console CLI downloads CRDs.

Reply:
Right, `oc-cli-downloads` contains the links to download the `oc` client from the download route:

```bash
$ oc get consoleclidownloads oc-cli-downloads -o yaml | yq .spec.links[].href
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/amd64/linux/oc.tar
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/amd64/mac/oc.zip
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/amd64/windows/oc.zip
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/arm64/linux/oc.tar
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/arm64/mac/oc.zip
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/ppc64le/linux/oc.tar
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/s390x/linux/oc.tar
https://downloads-openshift-console.apps.alebedev-0424.devcluster.openshift.com/oc-license
```

So, the risk is to have the broken links in `/command-line-tools` path of the console UI.

Reply:
@csrwng : can we accept the risk I described above?

Reply:
Yes

Reply:
Then we should be also prematurely exiting the DownloadsDeployment controller and [remove the deployment itself](https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/downloadsdeployment/controller.go#L129).

Reply:
[The new API](https://github.com/openshift/api/pull/1887) has been implemented to setup alternative ingress for the downloads.